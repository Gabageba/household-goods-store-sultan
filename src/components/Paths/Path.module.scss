.path {
  display: flex;
  align-items: center;
  margin-top: 44px;
  margin-bottom: 50px;

  &__line {
    height: 16px;
    margin-left: 10px;
    margin-right: 10px;
  }

  &__block {
    display: flex;
    align-items: center;
  }

  &__link {
    color: var(--sub-text-color);
    font-weight: 300;
  }

  &__currentLink {
    opacity: 0.5;
  }
}