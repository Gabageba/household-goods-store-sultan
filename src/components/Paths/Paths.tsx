import React, {FC} from 'react'
import styles from './Path.module.scss'
import {IPaths} from '../../types/path'
import {Link} from 'react-router-dom'
import {COSMETICS_HYGIENE_ROUTE} from '../../utils/consts'

interface PathProps {
  paths?: IPaths[]
}

const Paths: FC<PathProps> = ({paths}) => {
  return (
    <div className={styles.path}>
      <Link className={`${styles.path__link} ${!paths && styles.path__currentLink}`} to={COSMETICS_HYGIENE_ROUTE}>Главная</Link>
      {
        paths
          ? paths.map((path, id) =>
            <div className={styles.path__block}>
              <div className={`dashedLineVert ${styles.path__line}`}></div>
              <Link to={path.link} className={`${styles.path__link} ${id === paths.length - 1 && styles.path__currentLink}`}>{path.name}</Link>
            </div>
          )
          : ''
      }
    </div>
  )
}

export default Paths