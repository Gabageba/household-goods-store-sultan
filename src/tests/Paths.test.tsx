import {renderTestApp} from './helpers/renderTestApp'
import Paths from '../components/Paths/Paths'
import {screen} from '@testing-library/react'

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: '/cosmeticsHygiene/97450200003'
  })
}));

describe('path tests', () => {
  test('path test', () => {
    renderTestApp(<Paths/>)
    expect(screen.getByTestId('paths')).toMatchSnapshot()
  })
})