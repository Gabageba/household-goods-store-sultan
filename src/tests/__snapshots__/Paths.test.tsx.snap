// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`path tests path test 1`] = `
<div
  class="path"
  data-testid="paths"
>
  <div
    class="path__full"
  >
    <a
      class="path__link false"
      href="/cosmeticsHygiene"
    >
      Главная
    </a>
    <div
      class="path__block"
    >
      <div
        class="dashedLineVert path__line"
      />
      <a
        class="path__link path__currentLink"
        href="/cosmeticsHygiene"
      >
        Косметика и гигиена
      </a>
    </div>
  </div>
  <a
    class="path__back"
    href="/cosmeticsHygiene"
  >
    <div
      class="path__backButton"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 9 16"
        width="9"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 2.28571L3.375 8L9 13.7143L7.875 16L2.54292e-07 8L7.875 9.83506e-08L9 2.28571Z"
          fill="#FFC85E"
        />
      </svg>
    </div>
    <span>
      Назад
    </span>
  </a>
</div>
`;
